### ISSUE TITLE
How does this relate to flambo?

### ISSUE THREAD

**Issue by rizo** [gorillalabs/sparkling#1]
I found at least three versions of spark bindings on github:

- [gorillalabs/sparkling](https://github.com/gorillalabs/sparkling)
- [yieldbot/flambo](https://github.com/yieldbot/flambo)
- [chrisbetz/flambo](https://github.com/chrisbetz/flambo)

@yieldbot @chrisbetz
How are they related? What are the plans for development? Or basically, which one should I use and contribute to?

And also, will it be possible to integrate the cassandra specific code to one of them or it is better used as a separate dep: <https://github.com/codemomentum/flambo-cassandra>.

Thanks! ;)

**Comment by chrisbetz** [comment]
Hi RIzo,

thanks for asking - and sorry for the inconvenience of having to check out
three repos.

This is the line of things:

* In the beginning, there was clj-spark from "The Climate corp" (
https://github.com/TheClimateCorporation/clj-spark)

* From there, yieldlab developed flambo.

* I consolidated some forks in chrisbetz/flambo and during development, I
ran into serious performance issues:
  a) in serialization of functions
  b) by mapping back and forth from Clojure data structures to Scala Tuple2
objects. This looses the partitioner information in the RDDs (by design of
Spark), causing joins to result in (enormous) shuffles. That totally killed
my performance.

* As I made some serious changes to the API to overcome that problem, I
decided to run an independent project under gorillalabs/sparkling. So this
is identical to chrisbetz/flambo (which will be deleted sooner or later).
BUT: I'm in the middle of transitioning right now. The code is fine on
gorillalabs/sparkling and I'm using that in production. I'm still in the
middle of setting up github-pages for the project (see
https://gorillalabs.github.io/sparkling, which is work in progress right
now. I'm up to put a big "WIP" Sticker on top of it tomorrow until I
finished that). Same applies to streaming and SQL, which I do not use in
production as of now.


So, in short: You're happily invited to use and to contribute to
gorillalabs/sparkling.


Regarding the Cassandra-Repo. I'm not using Cassandra right now and I don't
like to have untested code in sparkling. But I will talk to
https://github.com/pesterhazy with whom I'm giving a talk in some days.
He's obviously interested also, so maybe that's a chance.

Hope that this clears up things!

Happy hacking

Chris

**Comment by chrisbetz** [comment]
Hi RIzo,

thanks for asking - and sorry for the inconvenience of having to check out
three repos.

This is the line of things:

* In the beginning, there was clj-spark from "The Climate corp" (https://github.com/TheClimateCorporation/clj-spark)
* From there, yieldlab developed flambo.
* I consolidated some forks in chrisbetz/flambo and during development, I ran into serious performance issues:
  * in serialization of functions
  * by mapping back and forth from Clojure data structures to Scala Tuple2 objects. This looses the partitioner information in the RDDs (by design of Spark), causing joins to result in (enormous) shuffles. That totally killed my performance.
* As I made some serious changes to the API to overcome that problem, I decided to run an independent project under gorillalabs/sparkling. So this is identical to chrisbetz/flambo (which will be deleted sooner or later).


BUT: I'm in the middle of transitioning right now. The code is fine on [gorillalabs/sparkling 1.0.0-SNAPSHOT] and I'm using that in production. I'm still in the middle of setting up github-pages for the project (see https://gorillalabs.github.io/sparkling, which is work in progress right now. I'm up to put a big "WIP" Sticker on top of it tomorrow until I finished that). Same applies to streaming and SQL, which I do not use in production as of now.


So, in short: You're happily invited to use and to contribute to gorillalabs/sparkling.


Regarding the Cassandra-Repo. I'm not using Cassandra right now and I don't like to have untested code in sparkling. But I will talk to https://github.com/pesterhazy with whom I'm giving a talk in some days. He's obviously interested also, so maybe that's a chance.

Hope that this clears up things!

Happy hacking

Chris

**Comment by chrisbetz** [comment]
Ok, I did it: We released gorillalabs/sparkling 1.0.0.

**Comment by rizo** [comment]
Thanks for the explanation and congratulations on the release! Excellent work!
